### PR TITLE
Fixes #35 #27

### LINKED ISSUE #35: Add map to detail art object view.
Detail art obejct activity should have a map. Location of art obejct need to be shown. This map need to be on the bottom of the detail view and represented as another card.

### PR DESCRIPTION
**Pull request by genaforvena**
Added map and gallery to the detail art object view.

Swapped map and list on the main fragment.

### CODE REVIEW

**Comment by genaforvena** [comment]
@sanovikov71 @oleg-popenov

File: app/src/main/java/org/imozerov/streetartview/ui/detail/DetailArtObjectActivity.kt [hunk 1]
```diff
@@ -42,21 +72,65 @@ class DetailArtObjectActivity : AppCompatActivity() {
...
+                    mapFragment?.getMapAsync {
+                        it.moveCamera(CameraUpdateFactory.newLatLngZoom(LatLng(artObjectUi.lat, artObjectUi.lng), 14f))
+                        it.addArtObject(artObjectUi)
+                    }
+                }
     }
 
-    companion object {
-        val EXTRA_KEY_ART_OBJECT_DETAIL_ID = "EXTRA_KEY_ART_OBJECT_DETAIL_ID"
+    internal inner class GalleryPagerAdapter(val context: Context, val images: List<String>) : PagerAdapter() {
```
Comment (sanovikov71) [inline app/src/main/java/org/imozerov/streetartview/ui/detail/DetailArtObjectActivity.kt:61]: Are you sure that using of an inner class that holds a reference to the outer class is safe here?

File: app/src/main/res/layout/activity_detail.xml [hunk 2]
```diff
@@ -20,20 +20,31 @@
             app:contentScrim="?attr/colorPrimary"
             app:layout_scrollFlags="scroll|exitUntilCollapsed">
 
-            <org.imozerov.streetartview.ui.detail.SquareImageView
-                android:id="@+id/art_object_detail_image"
```
Comment (sanovikov71) [inline app/src/main/res/layout/activity_detail.xml:24]: As I understood we do not have a collapsible toolbar now. Frankly said, that is a bit sad :)

File: app/src/main/res/values/strings.xml [hunk 3]
```diff
@@ -3,5 +3,7 @@
 
     <!-- TODO: Remove or change this placeholder text -->
```
Comment (sanovikov71) [inline app/src/main/res/values/strings.xml]: Does this comment make sense?

File: app/src/main/res/values/strings.xml [hunk 4]
```diff
@@ -3,5 +3,7 @@
...
     <string name="hello_blank_fragment">Hello blank fragment</string>
+    <string name="map_fragment_pager_label">Map</string>
+    <string name="list_fragment_pager_label">List</string>
     <string name="art_object_detail_image">Art object detail image</string>
```
Comment (sanovikov71) [inline app/src/main/res/values/strings.xml]: It was a content description of the removed detail image. It looks that this string is unused now.

File: app/src/main/java/org/imozerov/streetartview/ui/detail/DetailArtObjectActivity.kt (hunk 1 above)
Comment (genaforvena) [inline app/src/main/java/org/imozerov/streetartview/ui/detail/DetailArtObjectActivity.kt:61]: Yes. Adapter will be GCed with Activity.

File: app/src/main/res/values/strings.xml (hunk 3 above)
Comment (genaforvena) [inline app/src/main/res/values/strings.xml]: I'll remove this.

File: app/src/main/res/values/strings.xml (hunk 4 above)
Comment (genaforvena) [inline app/src/main/res/values/strings.xml]: Done.

**Comment by sanovikov71** [comment]
+1

**Pull request merged**